src/middleware/blacklist_ips.rs

```diff
@@ -0,0 +1,51 @@
+//! Middleware that blocks requests from a list of given IPs
+
+use super::prelude::*;
+
+use std::collections::HashMap;
+use std::net::IpAddr;
+
+// Can't derive debug because of Handler.
+#[allow(missing_debug_implementations)]
+#[derive(Default)]
+pub struct BlockIps {
+    ips: Vec<IpAddr>,
+    handler: Option<Box<Handler>>,
+}
+
+impl BlockIps {
+    pub fn new(ips: Vec<IpAddr>) -> Self {
+        Self {
+            ips,
+            handler: None,
+        }
+    }
+}
+
+impl AroundMiddleware for BlockIps {
+    fn with_handler(&mut self, handler: Box<Handler>) {
+        self.handler = Some(handler);
+    }
+}
+
+impl Handler for BlockIps {
+    fn call(&self, req: &mut Request) -> Result<Response, Box<Error + Send>> {
+        if self.ips.contains(&req.remote_addr().ip()) {
+            let body = "Due to unusual activity from your IP, it has \
```

Comment:
let's not say that we have blocked the IP. let's just say "We are unable to process your request at this time, please open an issue at <link>" and can we give them some identifying info to file with? an ID would be great